samples: bluetooth: added throughput many_to_one sample

SolveigLunde opened the pull request:
Created new sample for showcasing how use of PAwR can give higher throughput in many-to-one communitcation than classical BLE connections. This was done to show customers an alternative way to set up communication with many devices when they got insufficient throughput

divipillai commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,239 @@
+     - up to 4
+
+Limitations and trade-offs
+===========================
```
This section could be moved towards the bottom, just before the dependencies section.

divipillai commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,239 @@
+.. note::
+   Before building the advertiser, set :kconfig:option:`CONFIG_BT_NUM_RSP_DEVICES` in ``periodic_adv_rsp/prj.conf`` to the number of synchronizer devices you plan to run.
+   If this value does not match your setup, the advertiser may appear stalled while waiting for unassigned or missing responders.
+
```
```suggestion
   If this value does not match your setup, the advertiser might appear stalled while waiting for unassigned or missing responders.
```

divipillai commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,239 @@
+This sample provides predefined configuration files for typical use:
+
+- ``periodic_adv_rsp/prj.conf`` — advertiser/host settings (slot count, buffer sizes, timing).
+- ``periodic_sync_rsp/prj.conf`` — synchronizer/responder settings (PAwR sync, payload size, buffers).
```
```suggestion
* :file:`periodic_adv_rsp/prj.conf` - Advertiser/host settings (slot count, buffer sizes, and timing).
* :file:`periodic_sync_rsp/prj.conf` - Synchronizer/responder settings (PAwR sync, payload size, and buffers).

```

divipillai commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,239 @@
+   west build -p always -b nrf54l15dk/nrf54l15/cpuapp 
+
+After building, flash each role to its target kit.
+
```
```suggestion
```

divipillai commented on CODEOWNERS:
```diff
@@ -476,6 +476,7 @@
 /samples/bluetooth/shorter_conn_intervals/ @nrfconnect/ncs-dragoon
 /samples/bluetooth/subrating/             @nrfconnect/ncs-dragoon
 /samples/bluetooth/throughput/            @nrfconnect/ncs-blenders
+/samples/bluetooth/throughput_many_to_one/ @nrfconnect/ncs-dragoon
```
Please update the codeowners file for .rst files as well for the doc team.

umapraseeda replied:
Please add another entry for the new RST [here](https://github.com/nrfconnect/sdk-nrf/pull/26038/files#diff-fcf14c4b7b34fe7a11916195871ae66a59be87a395f28db73e345ebdc828085bL618). /samples/bluetooth/throughput_many_to_one/*.rst @nrfconnect/ncs-dragoon-doc

divipillai commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,239 @@
+
+- Failed to set subevent data (err -13)
+  - Controller rejected PAwR subevent parameters (e.g., subevent index out of range).
+  - Ensure subevent index is modulo the configured number of subevents and data length fits.
```
```suggestion
This section lists some common errors and possible solutions to them.

* [LOSS] slot X missed response, retransmit:

  * The advertiser expected data from that slot for this event but did not receive it.
  * After the first real payload from a slot, retransmit hints may appear under RF loss.

* [CLAIM] Reclaimed idle slot X:

  * The advertiser freed a slot after several missed events. 
     Responders must rejoin.

* Failed to set subevent data (err -13):

  * The Controller rejected PAwR subevent parameters (for example, subevent index out of range).
  * Ensure the subevent index is modulo the configured number of subevents and the data length fits.

```

richabp commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,239 @@
+References
+*************
+
+For more about Periodic Advertising with Responses (PAwR), see the Bluetooth Core Specification:
```
Could this reference be linked?

richabp reviewed (changes requested):
Please add a Changelog entry for this sample.
Also, there is no doc build check in the PR.

nordicjm commented on samples/bluetooth/throughput_many_to_one/periodic_adv_rsp/Kconfig:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "Throughput many-to-one Sample"
```
```suggestion
menu "Throughput many-to-one Sample"
```

nordicjm commented on samples/bluetooth/throughput_many_to_one/periodic_adv_rsp/Kconfig:
```diff
@@ -0,0 +1,21 @@
+	  Number of scanner devices that will participate in the PAwR communication.
+	  This affects timing calculations.
+		
+source "Kconfig.zephyr"
```
```suggestion
endmenu

source "Kconfig.zephyr"
```

richabp reviewed (commented):
It seems that CI is not triggered, please check the issue.

eirget committed:
samples: bluetooth: added throughput many_to_one sample

Created new sample for showcasing how use
of PAwR can give higher throughput
in many-to-one communitcation
than classical BLE connections.
This was done to show customers
an alternative way to set up
communication with many devices
when they got insufficient throughput

Signed-off-by: Solveig Lunde <Solveig.Lunde@nordicsemi.no>
Signed-off-by: Eirin Berget <Eirin.Berge@nordicsemi.no>

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, click the 'edited' button above_</sup>
Build number: 19
### Inputs:
#### Sources:
sdk-nrf: PR head: [021246a58cf2507cba8956476f30890916f4b95a](https://github.com/nrfconnect/sdk-nrf/pull/26038/commits/021246a58cf2507cba8956476f30890916f4b95a)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [021246a58cf2507cba8956476f30890916f4b95a](https://github.com/nrfconnect/sdk-nrf/pull/26038/commits/021246a58cf2507cba8956476f30890916f4b95a)
> merge base: [80fcb7b1c13968745a103566b3ca1aa9c343c4a7](https://github.com/nrfconnect/sdk-nrf/commit/80fcb7b1c13968745a103566b3ca1aa9c343c4a7)
> target head (main): [80fcb7b1c13968745a103566b3ca1aa9c343c4a7](https://github.com/nrfconnect/sdk-nrf/commit/80fcb7b1c13968745a103566b3ca1aa9c343c4a7)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/80fcb7b1c13968745a103566b3ca1aa9c343c4a7..021246a58cf2507cba8956476f30890916f4b95a)</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (12)</summary>

```
CODEOWNERS
doc
│  ├── nrf
│  │  ├── releases_and_maturity
│  │  │  ├── releases
│  │  │  │  │ release-notes-changelog.rst
samples
│  ├── bluetooth
│  │  ├── throughput_many_to_one
│  │  │  ├── README.rst
│  │  │  ├── periodic_adv_rsp
│  │  │  │  ├── CMakeLists.txt
│  │  │  │  ├── Kconfig
│  │  │  │  ├── prj.conf
│  │  │  │  ├── sample.yaml
│  │  │  │  ├── src
│  │  │  │  │  │ main.c
│  │  │  ├── periodic_sync_rsp
│  │  │  │  ├── CMakeLists.txt
│  │  │  │  ├── prj.conf
│  │  │  │  ├── sample.yaml
│  │  │  │  ├── src
│  │  │  │  │  │ main.c
```
</details>

### Outputs:

#### Toolchain
Version: eb11c27a58
Build docker image: docker-dtr.nordicsemi.no/sw-production/ncs-build:eb11c27a58_5ea73affbf

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :white_medium_square: Toolchain - Skipped: existing toolchain is used
- :white_check_mark: Build twister
   - sdk-nrf test count: 2
- :white_check_mark: Integration tests

<details><summary>Disabled integration tests</summary>

- 
   - test-fw-nrfconnect-nrf_lrcs_mosh
   - test-fw-nrfconnect-nrf_lrcs_positioning
   - desktop52_verification
   - test_ble_nrf_config
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_mesh
   - test-fw-nrfconnect-ble_samples
   - test-fw-nrfconnect-chip
   - test-fw-nrfconnect-fem
   - test-fw-nrfconnect-nfc
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_thingy91
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-nrf_crypto
   - test-fw-nrfconnect-ps-main
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-rs
   - test-fw-nrfconnect-tfm
   - test-fw-nrfconnect-thread-main
   - test-low-level
   - test-sdk-audio
   - test-sdk-dfu
   - test-sdk-find-my
   - test-sdk-mcuboot
   - test-sdk-wifi
   - test-secdom-samples-public
</details>



*Note: This message is automatically posted and updated by the CI*

divipillai commented on samples/bluetooth/throughput_many_to_one/README.rst:
```diff
@@ -0,0 +1,260 @@
+
+* Same-slot collisions - If two unassigned responders choose the same open slot in the same subevent, their claim responses collide at the PHY and the advertiser receives neither.
+  No acknowledgment (ACK) is issued that event.
+  Both devicesremain unassigned and retry in later events with probabilistic backoff.
```
```suggestion
  Both devices remain unassigned and retry in later events with probabilistic backoff.
```

divipillai reviewed (commented):
Could add a changelog entry for this PR change.
